IBM/ibm-common-service-operator
# port cert-rotation from Bedrock cert-manager

How to test:

Setting up dev cs-operator image on your cluster:

1. ensure you have future catalogsource on your cluster
2. export REGISTRY="<your registry here, e.g. quay.io/henry_h_li>"
3. run `make build-dev-image`
4. edit manager.yaml `image: quay.io/henry_h_li/common-service-operator-amd64:dev`
5. run `make bundle-manifests`
6. run `make build-bundle-image`
7. run `make run-bundle`

Testing:
1. create certificate chain with CA certificate and leaf certificate
2. delete CA secret
3. leaf secret should get re-created, check creation time/duration
4. either install a Bedrock service and delete its certificate secret or create a deployment to use leaf certificate secret from step 1
5. pod should restart when leaf secret is deleted
